src/GHC/Real_LHAssumptions.hs

```diff
@@ -21,11 +21,15 @@ class (GHC.Internal.Real.Real a, GHC.Internal.Enum.Enum a) => GHC.Internal.Real.
   GHC.Internal.Real.rem :: x:a -> y:{v:a | v /= 0} -> {v:a | ((v >= 0) && (v < y))}
   GHC.Internal.Real.mod :: x:a -> y:{v:a | v /= 0} -> {v:a | v = x mod y && ((0 <= x && 0 < y) => (0 <= v && v < y))}
 
-  GHC.Internal.Real.div :: x:a -> y:{v:a | v /= 0} -> {v:a | (v = (x / y)) &&
+  GHC.Internal.Real.div :: x:a -> y:{v:a | v /= 0} -> {v:a | (v = div x y) &&
                                                     ((x >= 0 && y >= 0) => v >= 0) &&
-                                                    ((x >= 0 && y >= 1) => v <= x) && 
-                                                    ((1 < y)            => v < x ) && 
-                                                    ((y >= 1)           => v <= x)  
+                                                    ((x >= 0 && y >= 1) => v <= x) &&
+                                                    ((1 < y && x >= 0)  => v < x) &&
+                                                    ((1 < y && x < 0)   => v > x) &&
+                                                    ((y >= 1)           => v <= x) &&
```

Comment:
`((y >= 1)           => v <= x)`: this is true only if `x` is non-negative.

Reply:
Fixed. Thanks!